packages/shared/src/components/modals/bookmark/MoveBookmarkModal.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { type ReactElement } from 'react';
+import { Modal, type ModalProps } from '../common/Modal';
+import { ModalHeader } from '../common/ModalHeader';
+import { Button, ButtonVariant } from '../../buttons/Button';
+import { BookmarkIcon, PlusIcon, VIcon } from '../../icons';
+import {
+  useBookmarkFolderList,
+  useCreateBookmarkFolder,
+} from '../../../hooks/bookmark';
+import { useLazyModal } from '../../../hooks/useLazyModal';
+import { LazyModal } from '../common/types';
+import type { BookmarkFolder } from '../../../graphql/bookmarks';
+import { useToastNotification } from '../../../hooks';
+import { FolderIcon } from '../../icons/Folder';
+
+type MoveBookmarkFolderModalProps = Omit<ModalProps, 'children'> & {
+  listId?: string;
+  postId: string;
+};
+
+const MoveBookmarkModal = ({
+  postId,
+  listId,
+  ...props
+}: MoveBookmarkFolderModalProps): ReactElement => {
+  const { displayToast } = useToastNotification();
+  const { createFolder } = useCreateBookmarkFolder();
+  const { openModal, closeModal } = useLazyModal();
+  const { folders, isPending, onMoveBookmark, isMoving } =
+    useBookmarkFolderList();
+
+  const handleMoveBookmark = (targetList?: string) => {
+    if (isMoving) {
+      return;
+    }
+    onMoveBookmark({ postId, listId: targetList });
+    displayToast(
+      `✅ Moved to ${
+        folders?.find((f) => f.id === targetList)?.name || 'Quick saves'
+      }`,
+      {
+        onUndo: () => handleMoveBookmark(listId),
+      },
+    );
+    closeModal();
+  };
+
+  const onCreateNewFolder = async (folder: BookmarkFolder) => {
+    const newFolder = await createFolder(folder);
+    handleMoveBookmark(newFolder.id);
+    closeModal();
+  };
+
+  const onClickCreateNewFolder = () => {
+    openModal({
```

Comment:
Does this work?
I know we had issues with lazy modal on top of lazy modal due to context only knowing 1

Reply:
It does, but have to make sure that closing the new one re-opens the old one. That's why I added the "onAfterClose" callback